addons/product/tests/test_product_attribute_value_config.py

```diff
@@ -399,6 +399,60 @@ def test_get_first_possible_combination(self):
         with self.assertRaises(StopIteration):
             next(gen)
 
+        # Making sure it's not extremely slow (has to discard invalid combinations early !)
+        product_template = self.env['product.template'].create({
+            'name': 'many combinations',
+        })
+
+        first_attr_first_value = None
+        second_attr_first_value = None
+        for i in range(10):
+            # create the attributes
+            product_attribute = self.env['product.attribute'].create({
+                'name': "att %s" % i,
+                'create_variant': 'dynamic',
+                'sequence': i,
+            })
+
+            for j in range(50):
+                # create the attribute values
+                value = self.env['product.attribute.value'].create([{
+                    'name': "val %s" % j,
+                    'attribute_id': product_attribute.id,
+                    'sequence': j,
+                }])
+
+            # set attribute and attribute values on the template
+            self.env['product.template.attribute.line'].create([{
+                'attribute_id': product_attribute.id,
+                'product_tmpl_id': product_template.id,
+                'value_ids': [(6, 0, product_attribute.value_ids.ids)]
+            }])
+
+        # Need to manually trigger compute as dependency are not specified, it's not the purpose of this perf test
+        product_template._compute_valid_attributes()
+
+        self._add_exclude(
+            self._get_product_template_attribute_value(product_template.attribute_line_ids[0].value_ids[0],
+                                                       model=product_template),
+            self._get_product_template_attribute_value(product_template.attribute_line_ids[1].value_ids[0],
+                                                       model=product_template),
+            product_template)
+
+        combination = self.env['product.template.attribute.value']
+        for idx, ptal in enumerate(product_template.attribute_line_ids):
+            if idx != 1:
+                value = ptal.product_template_value_ids[0]
+            else:
+                value = ptal.product_template_value_ids[1]
+            combination += value
+
+        started_at = time.time()
+        self.assertEqual(product_template._get_first_possible_combination(), combination)
+        elapsed = time.time() - started_at
+        # It should be about instantaneous, 0.5 to avoid false positives
+        self.assertLess(elapsed, 0.5)
```

Comment:
I have done it in another test in this file, nobody ever complained. It's quite important to have it, and it is reliable when the bad case is "many hours", we can set an arbitrary high threshold that still covers what we intend without the test ever failing non deterministically.

Reply:
Indeed, in case of high threshold, we will catch the big performance changes, let's keep it then (at worst we will raise the time threshold).  I'm used to complains on runbot false positives and was probably overly critical, it's still a good idea to add a test preventing high performance changes on products (since the threshold is quite bigger than the expected time).